pkg/components/exporter/exporter.go

```diff
@@ -94,57 +57,62 @@ var (
 	nsInfoDashboardJson string
 )
 
-func NewDaemonSet(detail components.Detail, k *v1alpha1.Kepler) *appsv1.DaemonSet {
+// TODO:
+func NewDaemonSet(detail components.Detail, k *v1alpha1.KeplerInternal) *appsv1.DaemonSet {
 	if detail == components.Metadata {
 		return &appsv1.DaemonSet{
-			TypeMeta: metav1.TypeMeta{APIVersion: appsv1.SchemeGroupVersion.String(), Kind: "DaemonSet"},
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: appsv1.SchemeGroupVersion.String(),
+				Kind:       "DaemonSet",
+			},
 			ObjectMeta: metav1.ObjectMeta{
-				Name:      DaemonSetName,
-				Namespace: components.Namespace,
-				Labels:    labels,
+				Name:      k.DaemonsetName(),
+				Namespace: k.Namespace(),
+				Labels:    labels(k),
 			},
 		}
 	}
 
-	deployment := k.Spec.Exporter.Deployment
+	deployment := k.Spec.Exporter.Deployment.ExporterDeploymentSpec
+	image := k.Spec.Exporter.Deployment.Image
 	nodeSelector := deployment.NodeSelector
 	tolerations := deployment.Tolerations
+	port := deployment.Port
```

Comment:
for multiple kepler deployments, the `port` also has to be different. Users must ensure that.